app/components/blacklight/document/page_header_component.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module Blacklight
+  module Document
+    # Render the start over and prev/next displays
+    class PageHeaderComponent < Blacklight::Component
+      attr_reader :document, :blacklight_config, :search_context, :search_session
+
+      delegate :blacklight_config, to: :helpers
+
+      def initialize(document:, search_context:, search_session:)
+        super
+        @search_context = search_context
+        @search_session = search_session
+        @document = document
+      end
+
+      def render?
+        search_context.present? || search_session.present? || has_header_tools?
```

Comment:
Is there a reason we need this method?  Won't `search_session` always exist, which means this is always truthful?  See https://github.com/projectblacklight/blacklight/blob/a8b5a3d3a2e8243c8a875d40528eae5c106d83bd/app/controllers/concerns/blacklight/search_context.rb#L56-L61

Reply:
Actually present? works for empty hashes, so I changed it back.